akka-typed-tests/src/test/scala/akka/typed/scaladsl/persistence/ApiTest.scala

```diff
@@ -0,0 +1,239 @@
+package akka.typed.scaladsl.persistence
+
+import akka.typed
+
+import scala.concurrent.duration._
+import akka.typed.{ ActorRef, Behavior, ExtensibleBehavior, Signal }
+import akka.typed.scaladsl.{ ActorContext, TimerScheduler }
+
+import scala.concurrent.ExecutionContext
+
+class ApiTest {
+  object TypedPersistentActor {
+
+    sealed trait PersistentEffect[+Event]
+
+    case object PersistNothing extends PersistentEffect[Nothing]
+
+    case class PersistAnd[Event](event: Event, callback: Event ⇒ Unit) extends PersistentEffect[Event]
+
+    case class Persist[Event](event: Event) extends PersistentEffect[Event] {
+      def andThen(callback: Event ⇒ Unit) = PersistAnd(event, callback)
+    }
+
+  }
+
+  object Actor {
+    import TypedPersistentActor._
+
+    class PersistentBehavior[Command, Event, State] extends ExtensibleBehavior[Command] {
+      override def receiveSignal(ctx: typed.ActorContext[Command], msg: Signal): Behavior[Command] = ???
+      override def receiveMessage(ctx: typed.ActorContext[Command], msg: Command): Behavior[Command] = ???
+
+      def onRecoveryComplete(callback: (ActorContext[Command], State) ⇒ Unit): PersistentBehavior[Command, Event, State] = ???
+      def snapshot[Snapshot](
+        onState: State ⇒ Option[Snapshot]          = (_: State) ⇒ None,
+        onEvent: Event ⇒ Option[Snapshot]          = (_: Event) ⇒ None,
+        on:      (State, Event) ⇒ Option[Snapshot] = (_: State, _: Event) ⇒ None,
```

Comment:
another way would be to have a `PersistentActorContext` with a `snapshot()` method that can be called